code/modules/spacepods/spacepod.dm

```diff
@@ -0,0 +1,1287 @@
+	cargo_hold.max_w_class = 5		//fit almost anything
+	cargo_hold.max_combined_w_class = 0 //you can optimize your stash with larger items
+	speed_multiplier = pod_armor.speed
+
+	armorDesc()
+
+	for(var/turf/T in locs)
+		for(var/obj/I in T.contents)
+			if(istype(I, /obj/structure/spacepod_frame))
+				QDEL_NULL(I)
+
+
+/obj/spacepod/Destroy()
+	if(equipment_system.cargo_system)
+		equipment_system.cargo_system.removed(null)
+	QDEL_NULL(equipment_system)
+	QDEL_NULL(cargo_hold)
+	QDEL_NULL(cell)
+	if(loc)
+		loc.assume_air(cabin_air)
+		air_update_turf()
+	QDEL_NULL(cabin_air)
+	QDEL_NULL(internal_tank)
+	occupant_sanity_check()
+	if(pilot)
+		pilot.forceMove(get_turf(src))
+		Remove_Actions(pilot)
+		pilot.clear_alert("charge")
+		pilot.clear_alert("mech damage")
+		pilot = null
+	if(passengers)
+		for(var/mob/living/M in passengers)
+			M.forceMove(get_turf(src))
+			Remove_Actions(M)
+			passengers -= M
+	GLOB.spacepods_list -= src
+	GLOB.poi_list.Remove(src)
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/spacepod/proc/update_icons()
+	cut_overlays()
+	icon_state = pod_armor.icon_state
+	if(is_cloaking && cell.charge > 2)
+		icon_state = "pod_cloaked"
+		alpha = 50
+		return
+	else
+		alpha = 255
+		icon_state = pod_armor.icon_state
+	if(!pod_overlays)
+		pod_overlays = new/list(2)
+		pod_overlays[DAMAGE] = mutable_appearance(icon, "pod_damage")
+		pod_overlays[FIRE] = mutable_appearance(icon, "pod_fire")
+	if(!pod_paint_effect)
+		pod_paint_effect = new/list(4)
+		pod_paint_effect[LIGHT] = mutable_appearance(icon,icon_state = "LIGHTS")
+		pod_paint_effect[WINDOW] = mutable_appearance(icon,icon_state = "Windows")
+		pod_paint_effect[RIM] = mutable_appearance(icon,icon_state = "RIM")
+		pod_paint_effect[PAINT] = mutable_appearance(icon,icon_state = "PAINT")
+	overlays.Cut()
+
+	if(has_paint)
+		var/image/to_add
+		if(!isnull(pod_paint_effect[LIGHT]))
+			to_add = pod_paint_effect[LIGHT]
+			to_add.color = colors[LIGHT]
+			add_overlay(to_add)
+		if(!isnull(pod_paint_effect[WINDOW]))
+			to_add = pod_paint_effect[WINDOW]
+			to_add.color = colors[WINDOW]
+			add_overlay(to_add)
+		if(!isnull(pod_paint_effect[RIM]))
+			to_add = pod_paint_effect[RIM]
+			to_add.color = colors[RIM]
+			add_overlay(to_add)
+		if(!isnull(pod_paint_effect[PAINT]))
+			to_add = pod_paint_effect[PAINT]
+			to_add.color = colors[PAINT]
+			add_overlay(to_add)
+	if(obj_integrity <= round(max_integrity/2))
+		add_overlay(pod_overlays[DAMAGE])
+		if(obj_integrity <= round(max_integrity/4))
+			add_overlay(pod_overlays[FIRE])
+	if(equipment_system && equipment_system.weapon_system && equipment_system.weapon_system.overlay_icon)
+		var/mutable_appearance/weapon_overlay = mutable_appearance(icon, icon_state = equipment_system.weapon_system.overlay_icon)
+		add_overlay(weapon_overlay)
+	light_color = icon_light_color[icon_state]
+	bound_width = 64
+	bound_height = 64
+
+/obj/spacepod/bullet_act(obj/item/projectile/P)
```

Comment:
there a reason this isn't calling parent?

Reply:
fixed

Reply:
i rescind my comment